citadel_workspace_service/src/kernel/mod.rs

```diff
@@ -178,15 +183,21 @@ async fn payload_handler(
                 None => info!(target: "citadel","connection not found"),
             };
         }
-        InternalServicePayload::ConnectionFailure { .. } => {}
-        InternalServicePayload::MessageReceived { .. } => {}
-        InternalServicePayload::Disconnect { .. } => {}
+
+        InternalServicePayload::Disconnect { cid, uuid } => match connection_map.get_mut(&cid) {
```

Comment:
```suggestion
        InternalServicePayload::Disconnect { cid, uuid } => match connection_map.remove(&cid) {
```

Disconnection should be a multi-part process.

1. Remove the tx handle from the hashmap instead of get_mut.
2. Stop the rx handle (which is a spawned task. Although, it may do this automatically when the connection disconnects, so we can add logging to after the `while let Some` loop inside the rx task.
3. Send a disconnect request to the protocol via the `NodeRemote`.